src/thor/costmatrix.cc

```diff
@@ -78,40 +80,49 @@ float CostMatrix::GetCostThreshold(const float max_matrix_distance) {
 // Clear the temporary information generated during time + distance matrix
 // construction.
 void CostMatrix::clear() {
-  // Clear the target edge markings
-  for (auto& iter : *targets_) {
-    iter.second.clear();
-    iter.second.resize(0);
-    iter.second.shrink_to_fit();
+  // resize all relevant structures down to 20 locations
+  if (source_count_ > kMaxLocationCache) {
+    source_edgelabel_.resize(kMaxLocationCache);
+    source_edgelabel_.shrink_to_fit();
+    source_adjacency_.resize(kMaxLocationCache);
+    source_adjacency_.shrink_to_fit();
+    source_edgestatus_.resize(kMaxLocationCache);
+    source_edgestatus_.shrink_to_fit();
+  }
+  if (target_count_ > kMaxLocationCache) {
+    target_edgelabel_.resize(kMaxLocationCache);
+    target_edgelabel_.shrink_to_fit();
+    target_adjacency_.resize(kMaxLocationCache);
+    target_adjacency_.shrink_to_fit();
+    target_edgestatus_.resize(kMaxLocationCache);
+    target_edgestatus_.shrink_to_fit();
```

Comment:
I chose to shrink all containers to 20 locations each after each request. so the server isn't severely hurt if someone decides to send off a 1000 location request.